repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/PassNonDedicatedTenantTest.groovy

```diff
@@ -0,0 +1,53 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.filters.valkyrie
+
+import framework.ReposeValveTest
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+
+class PassNonDedicatedTenantTest extends ReposeValveTest {
+
+    def setupSpec() {
+        Map params = properties.getDefaultTemplateParams()
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/valkyrie/passnondedicatedtenant", params)
+
+        repose.start()
+
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort, 'origin service')
+    }
+
+    def "when handling a request for a non-dedicated tenant's resource, the request should be passed without modification"() {
+        given:
+        String tenantId = "12345"
+
+        when:
+        MessageChain mc = deproxy.makeRequest(
+                url: reposeEndpoint + "/" + tenantId + "/foo",
+                method: "GET",
+                headers: ["X-Tenant-Id": tenantId])
+
+        then:
+        mc.getHandlings().size() == 1
+        mc.getReceivedResponse().getCode().toInteger() == 200
```

Comment:
So #1 is somewhat tricky because... well, what does it mean for a client's request to be unmodified? Both Deproxy and Repose will modify the headers. I added some assertions that check more, but they won't fail if additional headers are added.

I think I've covered all of the cases you listed now.